akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala

```diff
@@ -800,15 +800,15 @@ private[akka] class ShardRegion(
     else {
       shards.get(id).orElse(
         entityPropsFactory match {
-          case Some(props) if !shardsByRef.values.exists(_ == id) ⇒
+          case Some(factory) if !shardsByRef.values.exists(_ == id) ⇒
             log.debug("Starting shard [{}] in region", id)
 
             val name = URLEncoder.encode(id, "utf-8")
             val shard = context.watch(context.actorOf(
               Shard.props(
                 typeName,
                 id,
-                props(id),
+                factory,
```

Comment:
this is where the Shard is created, so the id is the shardId, not the entityId

Reply:
Ouch